src/main/java/ch/njol/skript/expressions/ExprSentCommands.java

```diff
@@ -0,0 +1,160 @@
+	"Modifications will affect what commands show up for the player to tab complete. They will not affect what commands the player can actually run.",
+	"Adding new commands to the list is illegal behavior and will be ignored."
+})
+@Examples({
+	"on send command list:",
+		"\tset command list to command list where [input does not contain \":\"]",
+		"\tremove \"help\" from command list"
+})
+@Since("INSERT VERSION")
+@Events("send command list")
+public class ExprSentCommands extends SimpleExpression<String> {
+
+	static {
+		Skript.registerExpression(ExprSentCommands.class, String.class, ExpressionType.SIMPLE, "[the] [sent] [server] command[s] list");
+	}
+
+	private EvtPlayerCommandSend parent;
+	private Kleenean delay;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentEvent(PlayerCommandSendEvent.class))
+			return false;
+		Structure structure = getParser().getCurrentStructure();
+		if (!(structure instanceof EvtPlayerCommandSend))
+			return false;
+		delay = isDelayed;
+		parent = (EvtPlayerCommandSend) structure;
+		return true;
+	}
+
+	@Override
+	@Nullable
+	protected String[] get(Event event) {
+		if (delay != Kleenean.FALSE) {
+			Skript.error("Can't change the command list after the event has already passed");
+			return null;
+		}
+		if (!(event instanceof PlayerCommandSendEvent)) {
+			Skript.error("The 'command list' expression can only be used in a 'send command list' event");
+			return null;
+		}
+		return ((PlayerCommandSendEvent) event).getCommands().toArray(new String[0]);
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case REMOVE:
+			case REMOVE_ALL:
+			case DELETE:
+			case SET:
+			case RESET:
+				return new Class[]{String[].class};
+			case ADD:
+			default:
+				return null;
+		}
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		if (!(event instanceof PlayerCommandSendEvent))
+			return;
+
+		Collection<String> commands = ((PlayerCommandSendEvent) event).getCommands();
+
+		// short circuit if we're just clearing the list
+		if (mode == ChangeMode.DELETE || mode == ChangeMode.REMOVE_ALL) {
+			commands.clear();
+			return;
+		}
+
+		List<String> deltaCommands = (delta != null && delta.length > 0) ? new ArrayList<>(Arrays.asList((String[]) delta)) : new ArrayList<>();
+		switch (mode) {
+			case REMOVE:
+				commands.removeAll(deltaCommands);
+				break;
+			case SET:
+				// remove all completely new commands, as adding new commands to the commands collection is illegal behaviour
+				List<String> newCommands = new ArrayList<>(deltaCommands);
+				newCommands.removeAll(parent.getOriginalCommands());
+				deltaCommands.removeAll(newCommands);
+				commands.clear();
+				commands.addAll(deltaCommands);
+				break;
+			case RESET:
```

Comment:
Because of the other comments, resetting of these values won't be needed as another plugin or script may have already modified the values, so the reset values are not true.